tools/CreateRuleFabricBot/CreateRuleFabricBot/Helpers/CodeOwnersFile.cs

```diff
@@ -0,0 +1,86 @@
+﻿using CreateRuleFabricBot.Helpers;
+using OutputColorizer;
+using System.Collections.Generic;
+using System.IO;
+
+namespace CreateRuleFabricBot
+{
+    public static class CodeOwnersFile
+    {
+        public static List<CodeOwnerEntry> ParseFile(string filePathOrUrl)
+        {
+            string content;
+            Colorizer.Write("Retrieving file content from [Yellow!{0}]... ", filePathOrUrl);
+            content = FileHelpers.GetFileContents(filePathOrUrl);
+            Colorizer.WriteLine("[Green!Done]");
+
+            return ParseContent(content);
+        }
+
+        public static List<CodeOwnerEntry> ParseContent(string fileContent)
+        {
+            Colorizer.Write("Parsing CODEOWNERS table... ");
+            List<CodeOwnerEntry> entries = new List<CodeOwnerEntry>();
+            string line;
+            using (StringReader sr = new StringReader(fileContent))
+            {
+                while ((line = sr.ReadLine()) != null)
+                {
+                    // Does the line start with '# PRLabel: "label1", "label2"
+
+                    // Remove tabs and trim extra whitespace
+                    line = NormalizeLine(line);
+
+                    // Empty line, move on
+                    if (string.IsNullOrEmpty(line))
+                    {
+                        continue;
+                    }
+
+                    CodeOwnerEntry entry = new CodeOwnerEntry();
+
+                    // if we have the moniker in the line, parse the labels
+                    while (line.StartsWith('#') && entry.ProcessLabelsOnLine(line))
+                    {
+                        // We need to read the next line as we parsed this line
+                        line = sr.ReadLine();
+
+                        if (line == null)
+                        {
+                            break;
+                        }
+
+                        // Remove tabs and trim extra whitespace
+                        line = NormalizeLine(line);
+                    }
+
+                    // Empty line, move on
+                    if (string.IsNullOrEmpty(line))
```

Comment:
Should we allow empty links between the labels and the path?

Reply:
On top of this it looks like we don't support any lines that contain just simple comments which might cause some issues at some point if anyone wants to add other comments.

Reply:
You mean not supporting comments between these special comments and the line that contains the path?

I thought about supporting that, but it becomes so much more complex to write the parser... I would say we require those comments to NOT have a space between them and the path...

Reply:
We could essentially accept everything, empty line, comment, labels... up until we read a path. Then close out the entry.

Reply:
Done.